### ISSUE TITLE
plugin is not a subclass

### ISSUE THREAD

**Issue by stabvone** [Tib3rius/AutoRecon#150]
Hi Sir,

Having below error when trying to perform a scan.

Plugin "RPCDump" in rpcdump.py is not a subclass of either PortScan, ServiceScan, or Report.
Plugin "MiniImpacketShell" in smbclient.py is not a subclass of either PortScan, ServiceScan, or Report.
Plugin "SMBConnection" in smbclient.py is not a subclass of either PortScan, ServiceScan, or Report.

autorecon --version
AutoRecon v2.0.19

**Comment by Tib3rius** [comment]
Hi @stabvone,

Did you modify the RPCDump plugin at all? Also, "MiniImpacketShell" and "SMBConnection" aren't part of the default plugin set, so I assume these are plugins you have created? If you can share the code for them I can likely figure out what is going on.

The RPCDump plugin that comes with AutoRecon seems to be working fine for me, which is why I think maybe you've modified it?

- Tib3rius

**Comment by stabvone** [comment]
Hi @Tib3rius,

didnt make any modification to the script. It was actually a fresh installation. I have attached the plugin from /root/.config/autorecon.
[Archive.zip](https://github.com/Tib3rius/AutoRecon/files/8422969/Archive.zip)

Thank you sir!

**Comment by Tib3rius** [comment]
Sorry but what you're saying makes no sense. You can search the source code in this repo, there's no reference to "MiniImpacketShell" or "SMBConnection" anywhere:

https://github.com/Tib3rius/AutoRecon/search?q=SMBConnection&type=code
https://github.com/Tib3rius/AutoRecon/search?q=MiniImpacketShell&type=code

Even in your rpcdump.py code, the plugin is clearly a subclass of ServiceScan:

```python
from autorecon.plugins import ServiceScan

class RPCDump(ServiceScan):
```

How did you install AutoRecon, was it from this repo? When you say it was a fresh installation, are you sure there was never any custom version installed in the past?

One thing I did notice is you created the archive file on a Mac. Are you running AutoRecon on a Mac because that's not supported.

**Comment by stabvone** [comment]
i install Autorecon via the pip method. Nope, its installed in kali, just that i copied the file out. When i try to locate, i found other instance of rpcdump.py. it is in the impacket module. it could that it is calling the plugin elsewhere. any idea how to fix it?

`kali@kali:/usr/local/bin$ cat /opt/impacket/examples/rpcdump.py
#!/usr/bin/env python
# Impacket - Collection of Python classes for working with network protocols.
#
# SECUREAUTH LABS. Copyright (C) 2021 SecureAuth Corporation. All rights reserved.
#
# This software is provided under a slightly modified version
# of the Apache Software License. See the accompanying LICENSE file
# for more information.
#
# Description:
#   DCE/RPC endpoint mapper dumper.
#
# Author:
#   Javier Kohen
#   Alberto Solino (@agsolino)
#
# Reference for:
#   DCE/RPC.
#

from __future__ import division
from __future__ import print_function
import sys
import logging
import argparse

from impacket.http import AUTH_NTLM
from impacket.examples import logger
from impacket.examples.utils import parse_target
from impacket import uuid, version
from impacket.dcerpc.v5 import transport, epm
from impacket.dcerpc.v5.rpch import RPC_PROXY_INVALID_RPC_PORT_ERR, \
    RPC_PROXY_CONN_A1_0X6BA_ERR, RPC_PROXY_CONN_A1_404_ERR, \
    RPC_PROXY_RPC_OUT_DATA_404_ERR

class RPCDump:
    KNOWN_PROTOCOLS = {
        135: {'bindstr': r'ncacn_ip_tcp:%s[135]'},
        139: {'bindstr': r'ncacn_np:%s[\pipe\epmapper]'},
`

**Comment by Tib3rius** [comment]
Ok, I may have a fix, but I'll need you to test / confirm.

First you'll need to uninstall autorecon using `pip uninstall autorecon` or `pipx uninstall autorecon` (whichever you used). Then remove the ~/.config/AutoRecon directory.

Clone the following branch: `git clone --branch import-fix https://github.com/Tib3rius/AutoRecon`

cd into the cloned directory, and then install using either of the following commands:

```
pip install .
pipx install .
```

You should then be able to run "autorecon" normally and hopefully the import issue is resolved. Let me know if that works and I can get it merged into the main branch.

**Comment by stabvone** [comment]
@Tib3rius i did the above, after the installation is done, cant even launch autorecon. it seems that it is no longer there. 

kali@kali:~/AutoRecon$ pip install .
Defaulting to user installation because normal site-packages is not writeable
Processing /home/kali/AutoRecon
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: toml<0.11.0,>=0.10.2 in /usr/lib/python3/dist-packages (from autorecon==2.0.19) (0.10.2)
Requirement already satisfied: Unidecode<2.0.0,>=1.3.1 in /usr/lib/python3/dist-packages (from autorecon==2.0.19) (1.3.2)
Requirement already satisfied: appdirs<2.0.0,>=1.4.4 in /usr/lib/python3/dist-packages (from autorecon==2.0.19) (1.4.4)
Requirement already satisfied: colorama<0.5.0,>=0.4.4 in /usr/lib/python3/dist-packages (from autorecon==2.0.19) (0.4.4)
Building wheels for collected packages: autorecon
  Building wheel for autorecon (pyproject.toml) ... done
  Created wheel for autorecon: filename=autorecon-2.0.19-py3-none-any.whl size=537647 sha256=ae7cf5c503c1e63e23778131cd6bbaa20dbbe07e7243bfad188c66893186ae9a
  Stored in directory: /tmp/pip-ephem-wheel-cache-efmrvxk1/wheels/ff/4a/61/d185cdbc5fe36f69b64560259f6213ab9c6a7e09b4370530ee
Successfully built autorecon
Installing collected packages: autorecon
  WARNING: The script autorecon is installed in '/home/kali/.local/bin' which is not on PATH.                                                             
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.                                       
Successfully installed autorecon-2.0.19

**Comment by Tib3rius** [comment]
Fixed in 9d3e2e684cd61989b90293d94f3af05834bd1154.
